### PR TITLE
✨ Better i18n in Vue & a few touchups

### PR DESCRIPTION
**Pull request by HerrLevin**

### CODE REVIEW

File: app/Http/Resources/SettingsResource.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class SettingsResource extends JsonResource
+{
+    /**
+     * Transform the resource into an array.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return array
+     */
+    public function toArray($request) {
```
Comment (MrKrisKrisu) [inline app/Http/Resources/SettingsResource.php]: Please add the return type.

File: app/Http/Resources/UserSettingsResource.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class UserSettingsResource extends JsonResource
+{
+    /**
+     * Transform the resource into an array.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return array
+     */
+    public function toArray($request) {
```
Comment (MrKrisKrisu) [inline app/Http/Resources/UserSettingsResource.php]: Please add the return type.

File: app/Http/Resources/UserSettingsResource.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+namespace App\Http\Resources;
+
+use Illuminate\Http\Resources\Json\JsonResource;
+
+class UserSettingsResource extends JsonResource
+{
+    /**
+     * Transform the resource into an array.
+     *
+     * @param \Illuminate\Http\Request $request
```
Comment (MrKrisKrisu) [inline app/Http/Resources/UserSettingsResource.php]: import Request

File: app/Http/Resources/UserSettingsResource.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            'displayName'    => (string) $this->name,
+            'username'       => (string) $this->username,
+            'trainDistance'  => (float) $this->train_distance,
+            'trainDuration'  => (int) $this->train_duration,
+            'trainSpeed'     => (float) $this->averageSpeed,
+            'points'         => (int) $this->points,
+            'twitterUrl'     => $this->twitterUrl ?? null,
+            'mastodonUrl'    => $this->mastodonUrl ?? null,
+            'privateProfile' => (bool) $this->private_profile,
+            'role'           => $this->role,
```
Comment (MrKrisKrisu) [inline app/Http/Resources/UserSettingsResource.php:28]: Are you sure you want to add role to the ressource? Its just for us internal.

**Review by MrKrisKrisu — CHANGES_REQUESTED** [review]

File: app/Http/Resources/UserSettingsResource.php (hunk 4 above)
Comment (HerrLevin) [reply app/Http/Resources/UserSettingsResource.php:28]: No, the role is also used to show menu-items like "admin", etc. Therefor it's significant for the Vue-API.

**Comment by MrKrisKrisu** [comment]
changes also in #443... closing here to prevent merge conflicts

**Pull request closed without merging**